test/index.test.js

```diff
@@ -258,4 +262,71 @@ if (isWin32) {
       done();
     });
   });
+
+  describe('windows path with drive or UNC volume', function() {
+    it('should return parent dirname from absolute path with drive letter', function(done) {
+      expect(gp('C:/')).toEqual('C:/');
+      expect(gp('C:/.')).toEqual('C:/');
+      expect(gp('C:/*')).toEqual('C:/');
+      expect(gp('C:/./*')).toEqual('C:/.');
+      expect(gp('C://')).toEqual('C:/');
+      expect(gp('C://*')).toEqual('C:/');
+      expect(gp('C:/path/*.js')).toEqual('C:/path');
+
+      expect(gp('C:\\')).toEqual('C:/');
+      expect(gp('C:\\.')).toEqual('C:/');
+      expect(gp('C:\\*')).toEqual('C:/');
+      expect(gp('C:\\.\\*')).toEqual('C:/.');
+      expect(gp('C:\\\\')).toEqual('C:/');
+      expect(gp('C:\\\\*')).toEqual('C:/');
+      expect(gp('C:\\path\\*.js')).toEqual('C:/path');
+
+      done();
+    });
+
+    it('should return parent dirname from relative path with drive letter', function(done) {
+      expect(gp('C:')).toEqual('C:.');
+      expect(gp('C:.')).toEqual('C:.');
+      expect(gp('C:*')).toEqual('C:.');
+      expect(gp('C:./*')).toEqual('C:.');
+      expect(gp('C:.//')).toEqual('C:./');
+      expect(gp('C:.//*')).toEqual('C:./');
+      expect(gp('C:path/*.js')).toEqual('C:path');
+
+      expect(gp('C:.\\*')).toEqual('C:.');
+      expect(gp('C:.\\\\')).toEqual('C:./');
+      expect(gp('C:.\\\\*')).toEqual('C:./');
+      expect(gp('C:path\\*.js')).toEqual('C:path');
+
+      done();
+    });
+
+    it('should return parent dirname from UNC path', function(done) {
+      expect(gp('\\\\System07\\C$/')).toEqual('\\\\System07\\C$/');
+      expect(gp('\\\\System07\\C$/.')).toEqual('\\\\System07\\C$/');
+      expect(gp('\\\\System07\\C$/*')).toEqual('\\\\System07\\C$/');
+      expect(gp('\\\\System07\\C$/./*')).toEqual('\\\\System07\\C$/.');
+      expect(gp('\\\\System07\\C$//')).toEqual('\\\\System07\\C$/');
+      expect(gp('\\\\System07\\C$//*')).toEqual('\\\\System07\\C$/');
+      expect(gp('\\\\System07\\C$/path/*.js')).toEqual('\\\\System07\\C$/path');
+
```

Comment:
Does nodejs on Windows understand `/System07/C$/` as a UNC path? If so, we should be flipping those backslashes since we say that no `\` character is allowed in globs.